[Enterprise] Finally implement "Disable automatic update checks" gpo

htcfreek opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

This policy disables automatic update checks and shows a warning in the user interface.

![image](https://github.com/microsoft/PowerToys/assets/61519853/a737630d-c127-42a9-88cc-ea0580d610ca)

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x] **Closes:** #24363
- [x] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [x] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx => Issue #30652

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed
* Tested the behavior of the settings ui.
* I can't test the behavior of the updater code. This must be done by the reviewers.

htcfreek commented:
Notes for reviewers:
- I have no 100% go by @crutkas. He didn't answered my last comment in the issue. But I decided to implement this now.
- I wasn't able to test the behavior of the updater code. Can you please do this.
- The update for the end user docs is not done yet as I wait on a second PR.

htcfreek commented on src/settings-ui/Settings.UI/SettingsXAML/Views/GeneralPage.xaml:
```diff
@@ -175,6 +175,15 @@
+                        IsClosable="False"
+                        IsOpen="{x:Bind Mode=OneWay, Path=ViewModel.ShowPeriodicUpdateCheckGpoInformation}"
+                        IsTabStop="{x:Bind Mode=OneWay, Path=ViewModel.ShowPeriodicUpdateCheckGpoInformation}"
+                        Severity="Warning"
```
Maybe "Warning" is to important here and we want to switch it to "Informarion".

htcfreek commented:
Notes for reviewers:
- I have no 100% go by @crutkas. He didn't answered my last comment in the issue. But I decided to implement this now.
- I wasn't able to test the behavior of the updater code. Can you please do this.
- The update for the end user docs is not done yet as I wait on a second PR.
- Maybe the severity "Warning" is to important here and we want to switch it to "Informarion".

Jay-o-Way commented:
Seeing inconsistency in the various InfoBars (severety and text). Can we use the same code/UX in every place?

Jay-o-Way commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -3913,6 +3910,12 @@ Activate by holding the key for the character you want to add an accent to, then
+    <value>Please check manually at regular intervals for updates.</value>
+  </data>
+  <data name="GPO_PeriodicUpdateCheckIsDisabled.Title" xml:space="preserve">
+    <value>The system administrator has disabled the automatic update checks.</value>
```
I'd have to check - or somebody else needs to confirm - but there is more than one way to formulate this. It could also be "[this setting/auto-updates] is disabled by [system administrator/GPO]"...

htcfreek replied:
I will try to change wording.

htcfreek replied:
@Jay-o-Way 
I have checked all the different GPO info bar texts. And except from the generic version all of them starting with "The system administrator ...". So we should keep the wording here an align all the messages in a follow up PR. Maybe we can use the generic version on all places except the Run Plugin policy and the Update check policy.

![image](https://github.com/microsoft/PowerToys/assets/61519853/36bcc48e-9918-407c-9008-c4b23bd08dd1)

Jay-o-Way commented on src/settings-ui/Settings.UI/SettingsXAML/Views/GeneralPage.xaml:
```diff
@@ -175,6 +175,15 @@
+                        IsOpen="{x:Bind Mode=OneWay, Path=ViewModel.ShowPeriodicUpdateCheckGpoInformation}"
+                        IsTabStop="{x:Bind Mode=OneWay, Path=ViewModel.ShowPeriodicUpdateCheckGpoInformation}"
+                        Severity="Warning"
+                        Visibility="{Binding Mode=OneWay, Path=ViewModel.ShowPeriodicUpdateCheckGpoInformation, Converter={StaticResource BoolToVisibilityConverter}}" />
```
Please remove the Visibility property. It has no use for an InfoBar.
And is the Margin needed?

htcfreek replied:
The margin is a layout fix. And as I am not sure if the margin not leads to a strange layout if the Infobar is closed, I have added the visibility property.

htcfreek replied:
@Jay-o-way
I have removed the obsolete visibility property. Is has definitely no use.

Regarding the margin fix: We need it and it only moves the space from before the info bar to after it. I have made two shots. The first one without margin fix and the second with margin fix:

![image](https://github.com/microsoft/PowerToys/assets/61519853/f2b1e0cd-5612-450b-863d-d69bb456b704)

![image](https://github.com/microsoft/PowerToys/assets/61519853/68d5052b-d189-4c62-be8a-7b43111f528d)

htcfreek commented:
> Seeing inconsistency in the various InfoBars (severety and text). Can we use the same code/UX in every place?

Regarding the severity: I am not sure what @crutkas like to use. But he mentioned in the past that updates are important. So I decided to use a yellow bar to make the user stumble over it and see that he/she has to check manually. We can change it if @crutkas is okay with a blue one.

htcfreek commented on src/gpo/assets/PowerToys.admx:
```diff
@@ -1,11 +1,11 @@
     <target prefix="powertoys" namespace="Microsoft.Policies.PowerToys" />
   </policyNamespaces>
-  <resources minRequiredRevision="1.6"/><!-- Last changed with PowerToys v0.76.0 -->
+  <resources minRequiredRevision="1.7"/><!-- Last changed with PowerToys v0.78.0 -->
```
Revision for 0.78.0 has to be 1.8 or higher.

crutkas commented:
So this just blocks PowerToys from checking for a new update?  I don't agree with this approach as knowing there is a new version is very useful.  

I think focusing more on https://github.com/microsoft/PowerToys/issues/25717 and hammering out details here would be a better way to solve this problem.

htcfreek commented:
> So this just blocks PowerToys from checking for a new update?  

This blocks PowerToys from searching for updates automatically.

This has no user setting and is more for company deployment situations. It is only a gpo.

> I think focusing more on https://github.com/microsoft/PowerToys/issues/25717 and hammering out details here would be a better way to solve this problem.

I can close the PR if you don't like to implement this.

htcfreek commented:
@crutkas 
I updated the screenshot:
![image](https://github.com/microsoft/PowerToys/assets/61519853/dd0ccd49-ca2f-4f1a-9991-248a7f57dea7)

Only the worker for to check every x minutes in the background without user interaction is disabled. Not the update check feature in PT settings. (For clarification I updated the message text.)

jaimecbernardo commented:
Hi @htcfreek , This is a change that we don't want to be brought in. Checking for the update is something that we want in PowerToys as a way to try and keep users in the latest.
Thanks for opening the PR, but this a change that we can't accept at this moment.

htcfreek commented:
@jaimecbernardo 
Should I ...
1. Close the issue?
2. Close the PR?
3. Remove the repared code from code base as it will not be implemented?

jaimecbernardo commented:
@htcfreek, at one point we did discuss that it would be nice to have a GPO settings that would allow to calm down updates for around 3 releases. Although I'm not sure how this can be done. Can perhaps we turn this into something like no updates for x amount of time since last update? Capping it at 90 days?

htcfreek commented:
> @htcfreek, at one point we did discuss that it would be nice to have a GPO settings that would allow to calm down updates for around 3 releases. Although I'm not sure how this can be done. Can perhaps we turn this into something like no updates for x amount of time since last update? Capping it at 90 days?

The X releases GPO exists for blocking Action Center notifications for three releases. My question was more about should we remove code that we will never use. As I understand this GPO is not wanted an will never gets implemented.

jaimecbernardo commented:
You're right. Closing this PR. Thanks for opening it all the same :) 
Thinking removing that admx entry might make sense, but perhaps another PR?
